common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/StorageManifestDao.scala

```diff
@@ -2,35 +2,22 @@ package uk.ac.wellcome.platform.archive.common.storage.services
 
 import uk.ac.wellcome.platform.archive.common.bagit.models.BagId
 import uk.ac.wellcome.platform.archive.common.storage.models.StorageManifest
-import uk.ac.wellcome.storage.vhs.{EmptyMetadata, VersionedHybridStore}
 import uk.ac.wellcome.storage.{ReadError, StorageError}
+import uk.ac.wellcome.storage.vhs.{EmptyMetadata, VersionedHybridStore}
 
 class StorageManifestDao(
-  underlying: VersionedHybridStore[String, StorageManifest, EmptyMetadata]
+  vhs: VersionedHybridStore[String, StorageManifest, EmptyMetadata]
 ) {
+  def get(id: BagId): Either[ReadError, StorageManifest] = vhs.get(s"$id")
+
+  def put(storageManifest: StorageManifest): Either[StorageError, Unit] = {
 
-  def get(id: BagId): Either[ReadError, StorageManifest] =
-    underlying.get(id = id.toString)
+    val ifNotExisting = (storageManifest, EmptyMetadata())
+    val ifExisting = (o: StorageManifest, m: EmptyMetadata) => (o, m)
```

Comment:
Since this will discard the update, is it better to say we should never call “put()” on an already-stored manifest? And maybe throw if we try to store something different?

Reply:
I'm not sure what the interface will be yet - i'd like to keep the `StorageManifestDao` around and functionally identical - but just simplify things to make it easier to implement multiple versions.